### PR TITLE
Update graphics

### PR DESCRIPTION
**Pull request by jdolence**
## PR Summary

Adds the ability to provide a list of variables in the input file that get tagged for graphics output.  In support of this, also adds parsing of multiline fields in the input file.  Adds some documentation on graphics and some python scripts to read and plot data.  Finally, does some cleanup surrounding graphics.

Addresses #98 and #92.  Relevant for #32.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Comment by jdolence** [comment]
@forrestglines @jlippuner tagged you guys for review. @AndrewGaspar wanted to make you aware also since you asked about how to actually look at the data.

**Comment by jdolence** [comment]
let's merge #56 before this one as I just merged in that branch.

File: scripts/python/phdf.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+            if self.NumDims > 2:
+                zMask = (k<zo[0] or k>=zo[1])
+            for j in yRange:
+                if self.NumDims > 1:
+                    yMask = (j<yo[0] or j>=yo[1])
+                for i in xRange:
+                    xMask = (i<xo[0] or i>=xo[1])
+                    self.isGhost[index] = ( xMask or yMask or zMask )
+                    self.BlockIdx[index] = [k, j, i]
+                    index += 1
```
Comment (Yurlungur) [inline scripts/python/phdf.py:109]: I prefer helper functions to be below the constructor. Unless this is a method that can be called *instead* of a constructor, in which case it should be static.

Comment (jdolence) [reply scripts/python/phdf.py]: fixed.

**Comment by jdolence** [comment]
@jlippuner @forrestglines do you want to comment or should I just merge?

**Comment by jlippuner** [comment]
> @jlippuner @forrestglines do you want to comment or should I just merge?

Looking at it now

File: scripts/python/phdf.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+
+            If variable is a vector, each element of the returned
+            numpy array is a vector of that length.  
+
+            Default is to return a flat array of length TotalCells.
+            However if flatten is set to False, a 4D (or 5D if vector)
+            array is returned that has dimensions [NumBlocks, Nz, Ny,
+            Nx] where NumBlocks is the total number of blocks, and Nz,
+            Ny, and Nx are the number of cells in the z, y, and x
+            directions respectively.
```
[the remark this answers is not in the document]
Comment (jdolence) [reply scripts/python/phdf.py:56]: @nmsriram actually wrote that class.  @nmsriram?

File: docs/graphics.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Graphics
+
+Parthenon allows users to select which fields are captured in the HDF5 (```.phdf```) graphics dumps at runtime.  In the input file, include a ```<Graphics>``` block and list of variables, such as
+```
+<Graphics>
+variables = density, velocity, & # comments are still ok
+            energy               # notice the & continuation character
+                                 # for multiline lists
```
[the remark this answers is not in the document]
Comment (jdolence) [reply docs/graphics.md:8]: It's a reasonable discussion to have, certainly.  Now that you've opened the issue, I'm going to resolve this conversation for this PR.

**Comment by jdolence** [comment]
2 approvals...pulling the trigger.

**Pull request merged**